src/lib/zAuction.ts

```diff
@@ -87,8 +103,41 @@ export async function getBidsForNft(
 	return bids;
 }
 
+export async function getBidsListForNfts(
+	baseApiUri: string,
+	contract: string,
+	tokenId: string[],
+) {
+	const nftIdsArray: string[] = [];
+
+	for (let i = 0; i < tokenId.length; i++) {
+		// it builds the array of nftIds strings
+		nftIdsArray.push(getNftId(contract, tokenId[i]));
+	}
+
+	const nftIdsBody = { nftIds: nftIdsArray }; //builds the correct body to send
+
+	const endpoints = getApiEndpoints(baseApiUri);
+
+	const response = await fetch(endpoints.bidListEndpoint, {
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' },
+		body: JSON.stringify(nftIdsBody),
+	});
+
+	const jsonResponse = (await response.json()) as BidsListResponseDto[]; //awaits the json of the response
+
+	const bidsListMap = new Map <string, NftIdBidsOfListsDto[]>(); //mapping of tokenId as key and returned bids as value
```

Comment:
The Typescript implementation of `Map<>` can be kinda heavy handed here. You can use a simpler map instead by creating an interface as such:

```
interface TokenBids {
  [tokenId: string]: Bid[] | undefined
 }
 
 const bidsMap = {} as TokenBids;
 bidsMap[someTokenId] = { some bid }
 ```